src/main/java/org/jetbrains/research/refactorinsight/processors/CommitMiner.java

```diff
@@ -92,9 +98,59 @@ public void handle(String commitId, List<Refactoring> refactorings) {
     }
   }
 
+  /**
+   * Mines a single commit if the mining process does not take longer than
+   * 60 seconds.
+   *
+   * @param commit  to be mined
+   * @param map     refactorings map
+   * @param project the current project
+   */
+  public static void mineAtCommitTimeout(VcsCommitMetadata commit,
+                                         Map<String, RefactoringEntry> map,
+                                         Project project) {
+    GitService gitService = new GitServiceImpl();
+    GitHistoryRefactoringMiner miner = new GitHistoryRefactoringMinerImpl();
+    ExecutorService service = Executors.newSingleThreadExecutor();
+    Future<?> f = null;
+    final Repository repository;
+    try {
+      repository = gitService.openRepository(project.getBasePath());
+    } catch (Exception e) {
+      return;
+    }
+    try {
+      Runnable r = () -> {
+        miner.detectAtCommit(repository,
```

Comment:
Why we do not use [this method](https://github.com/tsantalis/RefactoringMiner/blob/39bc66358ae08d2719eec8ae0b31a5dac76253a9/src/org/refactoringminer/api/GitHistoryRefactoringMiner.java#L78)?